recipes/compute.rb

```diff
@@ -61,6 +61,13 @@
     not_if '/sbin/ppc64_cpu --smt 2>&1 | ' \
       'grep -E \'SMT is off|Machine is not SMT capable\''
   end
+when 'x86_64'
+  kernel_module 'kvm_intel' do
+    onboot true
+    reload false
+    options %w(nested=1)
+    check_availability true
+  end
```

Comment:
This will break if the system has an AMD processor since that uses the ``kvm_amd`` module instead. Also I'm not sure if that module has the same option or offers nested virtualization as well. FWIW the current cluster is all Intel, but I think it would be good to ensure this only works on Intel processors. One way you could do this is via this attribute: ``node['dmi']['processor']['manufacturer']``.

For now, I wouldn't worry about adding AMD support until we run into this.

Reply:
FWIW `kvm_amd` uses the exact same parameter for enabling nested virtualization:

```
root@olive:~# modinfo kvm_amd
filename:       /lib/modules/3.16.0-4-amd64/kernel/arch/x86/kvm/kvm-amd.ko
license:        GPL
author:         Qumranet
alias:          cpu:type:x86,ven*fam*mod*:feature:*00C2*
depends:        kvm
intree:         Y
vermagic:       3.16.0-4-amd64 SMP mod_unload modversions 
parm:           npt:int
parm:           nested:int
```

Reply:
If that's the case then I recommend we just go ahead and add that support and assume it works.